composer/algorithms/copypaste/copypaste.py

```diff
@@ -0,0 +1,403 @@
+            trainer = Trainer(
+                model=model,
+                train_dataloader=train_dataloader,
+                eval_dataloader=eval_dataloader,
+                max_duration="1ep",
+                algorithms=[algorithm],
+                optimizers=[optimizer]
+            )
+
+    """
+
+    def __init__(
+        self,
+        p=0.5,
+        max_copied_instances=None,
+        area_threshold=25,
+        padding_factor=0.5,
+        jitter_scale=(0.01, 0.99),
+        jitter_ratio=(1.0, 1.0),
+        p_flip=0.9,
+        bg_color=-1,
+        input_key: Union[str, int, Tuple[Callable, Callable], Any] = 0,
+        target_key: Union[str, int, Tuple[Callable, Callable], Any] = 1,
+    ):
+        self.input_key = input_key
+        self.target_key = target_key
+        self.configs = {
+            "p": p,
+            "max_copied_instances": max_copied_instances,
+            "area_threshold": area_threshold,
+            "padding_factor": padding_factor,
+            "jitter_scale": jitter_scale,
+            "jitter_ratio": jitter_ratio,
+            "p_flip": p_flip,
+            "bg_color": bg_color
+        }
+
+    def match(self, event: Event, state: State) -> bool:
+        return event == Event.AFTER_DATALOADER
+
+    def apply(self, event: Event, state: State, logger: Logger) -> None:
+        images = state.batch_get_item(key=self.input_key)
+        masks = state.batch_get_item(key=self.target_key)
+
+        out_images, out_masks = copypaste_batch(images, masks, self.configs)
+
+        state.batch_set_item(key=self.input_key, value=out_images)
+        state.batch_set_item(key=self.target_key, value=out_masks)
+
+
+def _copypaste_instance(src_image, src_mask, trg_image, trg_mask, src_instance_id, configs):
+    """Applies copy-paste augmentation on a set of source and target samples. The 
+    instance identified by ``src_instance_id`` is selected from the source sample 
+    to be copied to the target sample.
+
+    Args:
+        src_image (torch.Tensor): Source image of shape ``(C, H, W)``,
+            C is the number of channels.
+        src_mask (torch.Tensor): Source mask of shape ``(H, W)``,
+        trg_image (torch.Tensor): Target image of shape ``(C, H, W)``,
+            C is the number of channels.
+        trg_mask (torch.Tensor): Target mask of shape ``(H, W)``,
+        src_instance_id (int): Class ID of the randmoly chosen instance to be 
+            copied from the source sample into the target sample.
+        configs (dict): Configurable hyperparameters.
+
+    Returns:
+        trg_image (torch.Tensor): Augmented target image of shape ``(C, H, W)``,
+            C is the number of channels.
+        trg_mask (torch.Tensor): Augmented target mask of shape ``(H, W)``,
+    """
+    zero_tensor = torch.zeros(
+        1, dtype=src_image.dtype, device=src_image.device)
+    bg_color = configs["bg_color"]
+
+    src_instance_mask = _parse_mask_by_id(src_mask, src_instance_id, bg_color)
```

Comment:
```suggestion
    src_instance_mask = torch.where(src_mask == src_instance_id, src_instance_id, bg_color)
```
Maybe just inline the function since it is one line?